### ISSUE TITLE
Can not install geeklog 2.2.0 on PHP 5.3

### ISSUE THREAD

**Issue by jameswill** [Geeklog-Core/geeklog#877]
Hi,

I was trying to install Geeklog 2.2.0 on one of my servers with PHP 5.3. However I am getting following error.

**Fatal error: Can't use function return value in write context in {Path}/{To}/{geeklog220}/public_html/lib-common.php on line 7947**

On further debugging I found that it is trying to use function while returning the value in empty() function. However, prior to PHP 5.5, empty() only supports variables; anything else will result in a parse error.

Is the minimum requirement to install Geeklog has been changed to PHP 5.5 ? If yes then can anyone please update the documentation.

**Comment by eSilverStrike** [comment]
Thanks for the report and your debugging. Geeklog should still support PHP 5.3.3 and higher but has not been completely tested recently.

It looks like this bug only happens on multi-language sites. For line 7947 in the function COM_createHREFLang change:

`if (!empty(PLG_getItemInfo($type, $lang_item_id, 'id'))) { `
to
```
// In PHP 5.3 only variables can be used in empty
$retval = PLG_getItemInfo($type, $lang_item_id, 'id');
if (!empty($retval)) { 
```

That should fix the issue.

If you find any more issues let us know. If we run into too many other issues we may have to bump up the minimum PHP version.

**Comment by jameswill** [comment]
Hi,

Apologies for the late response.

The above code does solves the issue I stated above. However I have found same error on continuing the installation ahead on PHP 5.3

**Fatal error: Can't use function return value in write context in {Path}/{To}/{geeklog220}/plugins/calendar/functions.inc on line 704**

**Comment by eSilverStrike** [comment]
Thanks for letting us know. if you find any other related errors.

**Comment by jameswill** [comment]
Hi,

Shall we wait for any work around for PHP 5.3 or increase the minimum requirement to PHP 5.5 ?

**Comment by eSilverStrike** [comment]
We may end up increasing the minimum version, depends on how your testing goes (if you plan on testing anymore?) and if one of the other developers has a chance to test it with PHP 5.3.

**Comment by jameswill** [comment]
We have tried installing it on PHP 5.5 and its working fine. No issues were found as of now.

**Comment by jameswill** [comment]
Hi eSilverStrike,

Shall we wait for official notification for PHP 5.5 or shall we increase the minimum PHP requirement to PHP 5.5 and go ahead ?

**Comment by jameswill** [comment]
Hi,

Any update regarding this ?

**Comment by eSilverStrike** [comment]
I am able to test things at the moment. Did you find anymore PHP 5.3 errors in the install?

The above error you mentioned:

Fatal error: Can't use function return value in write context in {Path}/{To}/{geeklog220}/plugins/calendar/functions.inc on line 704

Seems to be in the function plugin_submit_calendar. This function is not run by the install so you got htis error after the install???

If these are the only 2 errors (or only a few more that we can fix easily enough) we will keep the min version at 5.3 else we will up the version to PHP 5.5 for Geeklog 2.2.1

**Comment by eSilverStrike** [comment]
Fixed spelling above it did say "able to test" it should say "unable to test"

**Comment by jameswill** [comment]
Hi,

No. I did not got any other error as I am unable install it successfully on PHP 5.3

I got another error while installing. After filling database details when I click on option **"Install (then select plugins to install)"** my installation page redirects and gives "The connection was reset" error on screen. and this error I am not getting on PHP 5.5

**Comment by eSilverStrike** [comment]
Okay then I have a feeling for Geeklog 2.2.1 we will push the PHP version to 5.5

**Comment by jameswill** [comment]
Hi,

Great. Is there any ETA for Geeklog 2.2.1 ?

**Comment by eSilverStrike** [comment]
It will be a bug fix release most likely for September.

**Comment by jameswill** [comment]
Hi,

Shall we wait till September or go ahead with this release with increasing minimum requirement to PHP 5.5 ?

**Comment by eSilverStrike** [comment]
We will increase the minimum but I can't do a official release until September as I am away most of the summer and do not have access to my development environment. I will post a warning on Geeklog.net that Geeklog 2.2.0 is only compatible with PHP 5.5 and higher.

**Comment by mystralkk** [comment]
Fixed with changeset bb78dd3.  This fix should make Geeklog 2.2.x compatible with PHP 5.3.
